### PR TITLE
blocky: init at 0.17

### PR DESCRIPTION
**Pull request by ratsclub**
###### Motivation for this change
Added [blocky](https://github.com/0xERR0R/blocky), a fast and lightweight DNS proxy as ad-blocker for local network with many features 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/blocky/default.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{ buildGo117Module, fetchFromGitHub }:
+
+buildGo117Module rec {
```
Comment (zowoq) [inline pkgs/applications/networking/blocky/default.nix:3]:
```suggestion
{ buildGoModule, fetchFromGitHub }:

buildGoModule rec {
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -24443,6 +24443,8 @@ with pkgs;
 
   bambootracker = libsForQt5.callPackage ../applications/audio/bambootracker { };
 
+  blocky = callPackage ../applications/networking/blocky { };
```
Comment (zowoq) [inline pkgs/top-level/all-packages.nix:24446]:
```suggestion
  blocky = callPackage ../applications/networking/blocky {
    buildGoModule = buildGo117Module;
  };
```

**Review by zowoq — CHANGES_REQUESTED** [review]
`meta` needs to be set.

File: pkgs/applications/networking/blocky/default.nix [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-vG6QAI8gBI2nLRQ0nOFWQHihyzgmJu69rgkWlg3iW3E=";
+  };
+
+  # needs network connection and fails at
+  # https://github.com/0xERR0R/blocky/blob/development/resolver/upstream_resolver_test.go
+  doCheck = false;
+
+  vendorSha256 = "sha256-my+yxDgKxoQvNhQ13T5m9pCWAIOUkdQixvKEZIaCMo0=";
```
Comment (jonringer) [inline pkgs/applications/networking/blocky/default.nix:18]:
```suggestion
  vendorSha256 = "sha256-+mpNPDejK9Trhw41SUXJPL/OX5wQR0QfA2+BXSlE0Jk=";
```

```
error: hash mismatch in fixed-output derivation '/nix/store/xda02qhz81arrh24hv2y1fs56nma73wd-blocky-0.17-go-modules.drv':
         specified: sha256-my+yxDgKxoQvNhQ13T5m9pCWAIOUkdQixvKEZIaCMo0=
            got:    sha256-+mpNPDejK9Trhw41SUXJPL/OX5wQR0QfA2+BXSlE0Jk=

**Comment by aanderse** [comment]
@ratsclub you should probably add a NixOS module to make this easier for people to use. Please feel free to copy/paste and finish [this](https://github.com/NixOS/nixpkgs/compare/master...aanderse:nixos/blocky?expand=1). Let me know if you need anything.

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (ratsclub) [reply pkgs/top-level/all-packages.nix:24446]: Thanks!

File: pkgs/applications/networking/blocky/default.nix (hunk 1 above)
Comment (ratsclub) [reply pkgs/applications/networking/blocky/default.nix]: Thanks!

File: pkgs/applications/networking/blocky/default.nix (hunk 3 above)
Comment (ratsclub) [reply pkgs/applications/networking/blocky/default.nix]: Thanks!

**Review by fpletz — APPROVED** [review]

**Comment by ratsclub** [comment]
@aanderse may I do this in another PR?

**Pull request merged**